### PR TITLE
feat(hostinfo/args): Allow hostinfos to be passed args

### PR DESCRIPTION
**Pull request by kaustavha**

### CODE REVIEW

**Comment by kaustavha** [comment]
Still gotta muck about in ele

File: hostinfo/read.lua [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  local outTable, errTable = {}, {}
+
+  local function finalCb(err, data)
+    misc.safeMerge(errTable, err)
+    misc.safeMerge(outTable, data)
+    self:_pushParams(errTable, outTable)
+    return callback()
+  end
+
+  local type = fs.statSync(name).type
```
Comment (rphillips) [inline hostinfo/read.lua]: need to check for err

File: hostinfo/read.lua [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  if type == 'directory' then
+    fs.readdir(name, finalCb)
+  elseif type == 'file' then
+    fs.readFile(name, finalCb)
```
Comment (rphillips) [inline hostinfo/read.lua]: the finalCb isn't called if there is an error

Comment (kaustavha) [inline hostinfo/read.lua]: Fixed.

**Comment by kaustavha** [comment]
Note: For read, whitelist allowed directories.

**Comment by kaustavha** [comment]
Related ele pr: https://github.com/racker/ele/pull/3150

**Comment by kaustavha** [comment]
May be superseded by the super-agent https://github.com/virgo-agent-toolkit/super-agent

**Pull request closed without merging**